mempool/src/shared_mempool/tasks.rs

```diff
@@ -151,6 +153,30 @@ pub(crate) async fn process_client_transaction_submission<NetworkClient, Transac
     }
 }
 
+/// Processes request for all addresses in parking lot
+pub(crate) async fn process_parking_lot_addresses<NetworkClient, TransactionValidator>(
+    smp: SharedMempool<NetworkClient, TransactionValidator>,
+    callback: oneshot::Sender<Vec<(AccountAddress, usize)>>,
+    timer: HistogramTimer,
+) where
+    NetworkClient: NetworkClientInterface<MempoolSyncMsg>,
+    TransactionValidator: TransactionValidation + 'static,
+{
+    let addresses = smp.mempool.lock().get_parking_lot_addresses();
+
+    //TODO @hariria check timer location
+    timer.stop_and_record();
```

Comment:
These should be at the beginning of the function. That being said, I don't think the task start timers are really useful. We should probably just remove them all.